File: pedidos-app/src/app/services/pedido.model.ts
export interface Comprador {
  nome: string;
}

export interface Fornecedor {
  nome: string;
}

export interface Produto {
  nome: string;
}

export interface ItemPedido {
  pedidoId: number;
  produto: Produto;
  quantidade: number;
  valorTotalProduto: number;
}

export interface Pedido {
  id: number;
  comprador: Comprador;
  fornecedor: Fornecedor;
  totalProdutos: number;
  valorTotal: number;
}

File: pedidos-api/src/main/java/br/com/tech/pedidos/api/model/PedidoModel.java
package br.com.tech.pedidos.api.model;

import br.com.tech.pedidos.domain.model.Comprador;
import br.com.tech.pedidos.domain.model.Fornecedor;
import br.com.tech.pedidos.domain.model.ItemPedido;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
public class PedidoModel {
    private Long id;
    private Comprador comprador;
    private Fornecedor fornecedor;
    private ItemPedido itemPedido;
    private Integer totalProdutos;
    private BigDecimal valorTotal;
}
